### PR TITLE
Add the latest version of oracledb to api/package.json

### PR DESCRIPTION
**Pull request by stevefan1999-personal**
~~Judging from https://github.com/directus/directus/actions/runs/5424640097/jobs/9864290720, there is only one unit test not being able to pass. Looks like we can just add it to the mix now.~~ This is not correct and it seems like the version is still pinned to 5.

I need this because trying to install NPM package from Docker image externally is very painful, here's a workaround I tried:

```
ARG VERSION

FROM directus/directus:${VERSION}

USER root
RUN chmod 777 -R /usr/local/bin/
WORKDIR /directus
RUN corepack enable && corepack prepare

RUN pnpm install oracledb --save
USER node
```

Clearly it is not elegant since it will bypass the `pnpm pack`:

https://github.com/directus/directus/blob/816f7dc00a4b94250f11086b71440fd83c9fcb38/Dockerfile#L14-L28

Let alone the dependencies should be frozen and never to be updated again.

### CODE REVIEW

**Comment by stevefan1999-personal** [comment]
I can't regenerate pnpm lockfile, please do this for me

**Comment by stevefan1999-personal** [comment]
Seeks like we are still using oracledb 5.x.x, hmm so the test result is not representative...

**Comment by rijkvanzanten** [comment]
The node Oracledb package can't be installed under ARM platforms (eg any modern mac), so this is not something we can add to the dependencies section of the package.json file unfortunately

**Comment by stevefan1999-personal** [comment]
> The node Oracledb package can't be installed under ARM platforms (eg any modern mac), so this is not something we can add to the dependencies section of the package.json file unfortunately

The latest 6.x version of oracledb implemented a pure JS driver which I think should be out of the box compatible. I did not see any breaking changes or migration guide so I assumed they have backward compatibility.

**Comment by rijkvanzanten** [comment]
> The latest 6.x version of oracledb implemented a pure JS driver which I think should be out of the box compatible. I did not see any breaking changes or migration guide so I assumed they have backward compatibility.

Fantastic!! That's news to me 😄 Glad to hear that incredibly annoying requirement on native binaries that may or may not exist is a thing of the past. In that case, forget what I said before 🙂

File: .github/workflows/blackbox-main.yml
Comment (rijkvanzanten) [inline .github/workflows/blackbox-main.yml:49]: I think the whole "Install Oracle client" block can be removed, as the dependency will exist in the `api`s package.json and is therefore installed in the regular install step

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]

**Comment by stevefan1999-personal** [comment]
Now we need to run the blackbox test...preferrably before merging the PR.

**Comment by stevefan1999-personal** [comment]
```
✖ Bootstrap databases and start servers [FAILED: Directus-oracle bootstrap failed: 
 Error: select * from "directus_fields" where "interface" = :1 - ORA-01002: fetch out of sequence
    at Protocol._processMessage (/home/runner/work/directus/directus/dist/node_modules/.pnpm/oracledb@6.0.2/node_modules/oracledb/lib/thin/protocol/protocol.js:167:17)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async ThinConnectionImpl.execute (/home/runner/work/directus/directus/dist/node_modules/.pnpm/oracledb@6.0.2/node_modules/oracledb/lib/thin/connection.js:802:7)
    at async Connection.execute (/home/runner/work/directus/directus/dist/node_modules/.pnpm/oracledb@6.0.2/node_modules/oracledb/lib/connection.js:861:16)
    at async Connection.<anonymous> (/home/runner/work/directus/directus/dist/node_modules/.pnpm/oracledb@6.0.2/node_modules/oracledb/lib/util.js:162:14) {
  offset: 0,
  errorNum: 1002,
  code: 'ORA-01002'
}
]
```

Hmm...[ORA-01002: fetch out of sequence tips (dba-oracle.com)](http://www.dba-oracle.com/t_ora_01002_fetch_out_of_sequence.htm)

**Comment by stevefan1999-personal** [comment]
https://github.com/stevefan1999-personal/directus/blob/8c23245b47394a978888e7c502630c691666c0ab/api/src/database/migrations/20210802A-replace-groups.ts#L6-L7
I'm not sure if this is the offending statement...

**Review by rijkvanzanten — APPROVED** [review]
Change LGTM, but lets see if the test will pass now (no clue why it shouldn't after this change, but oracle has surprised me in the past)

**Comment by rijkvanzanten** [comment]
Hmm same problem again. @aidenfoxx ever seen this issue fly by before? Kinda surprised it only started throwing this after upgrading the connection library

**Comment by aidenfoxx** [comment]
I've not seen it before. Could be a race condition, if other migrations are still being brought up on the Oracle side for some reason? Only think I can think of that would cause such a basic query to fail. 🤷

**Comment by licitdev** [comment]
I wonder if the error is due to mismatched package versions, as Knex is using `oracledb@5.5.0`, pending a dependency bump. 🤔 

Ref: https://github.com/knex/knex/pull/5630, https://github.com/knex/knex/issues/5615

**Comment by stevefan1999-personal** [comment]
YES! I've been monitoring the blackbox test for Oracle under the Actions tab, and some of the tests started passing.

It seems like https://github.com/oracle/node-oracledb/releases/tag/v6.1.0 and https://github.com/oracle/node-oracledb/issues/1565 was indeed the blocking factor and now it is fixed a week ago now, we can finally get this PR moving on

**Comment by stevefan1999-personal** [comment]
All cleared.
```
Test Suites: 36 passed, 36 total
Tests:       9968 passed, 9968 total
Snapshots:   0 total
Time:        1778.646 s
Ran all test suites.
❯ Stop Directus servers
❯ OracleDB
✔ OracleDB
✔ Stop Directus servers
```
@rijkvanzanten I think it is good. Let's merge it!

**Comment by stevefan1999-personal** [comment]
@[paescuj](https://github.com/paescuj) sorry to bother but do you think we need to have a way to let PRs to test the main workflow so that they can be safely merged. I think we can limit certain action workflows to get permission from maintainers first before launching.

**Comment by paescuj** [comment]
> It seems like https://github.com/oracle/node-oracledb/releases/tag/v6.1.0 and https://github.com/oracle/node-oracledb/issues/1565 was indeed the blocking factor and now it is fixed a week ago now, we can finally get this PR moving on.

Good news 😃 

> @rijkvanzanten I think it is good. Let's merge it!

Since we're using `oracledb` via Knex, we should wait for official support (https://github.com/knex/knex/issues/5615). This should land soon, Node.js 20 will be LTS from October 24.

> @[paescuj](https://github.com/paescuj) sorry to bother but do you think we need to have a way to let PRs to test the main workflow so that they can be safely merged. I think we can limit certain action workflows to get permission from maintainers first before launching.

I don't think this is necessary, it's usually sufficient to run only the SQLite test in PRs. Exceptions are rare and can be handled manually 👍

**Comment by stevefan1999-personal** [comment]
@paescuj 

> Since we're using `oracledb` via Knex, we should wait for official support ([knex/knex#5615](https://github.com/knex/knex/issues/5615)). This should land soon, Node.js 20 will be LTS from October 24.

My concern is that the upstream Knex are trying to upgrade the version to 6.1.0 (as evidenced here: https://github.com/knex/knex/pull/5675). The workflow for Oracle tests seems fine, but if you look closely, test errors are deliberately ignored. From my own prediction I'd be safe to assume at least 10% of them are supposed to be failing. Still, our integration tests here at Directus shows there is no problem whatsoever (all integration tests with Oracle Driver 6.1.0 passed, with Knex version compatibility bypassed thanks to semver).

**Comment by stevefan1999-personal** [comment]
@paescuj [Bump oracledb from 5.5.0 to 6.1.0 by dependabot\[bot\] · Pull Request #5675 · knex/knex (github.com)](https://github.com/knex/knex/pull/5675) is merged.

**Comment by rijkvanzanten** [comment]
heyyyy exciting stuff! So just to confirm, we're awaiting a new Knex release and then it's time to ship it?! ⛵

**Comment by paescuj** [comment]
> heyyyy exciting stuff! So just to confirm, we're awaiting a new Knex release and then it's time to ship it?! ⛵

That would be my preferred way ⛵😃

**Comment by stevefan1999-personal** [comment]
**@paescuj** they have released 3.0.0 which included oracledb 6.1.0, mind merging this?

**Comment by paescuj** [comment]
> **@paescuj** they have released 3.0.0 which included oracledb 6.1.0, mind merging this?

Yeah, I'll look into this soon 👍

**Review by paescuj — APPROVED** [review]
Thanks @stevefan1999-personal!

**Comment by stevefan1999-personal** [comment]
@paescuj much appreciated for the follow ups!

**Pull request merged**
